week3/1807_evaluate_bracket_pairs.py

```diff
@@ -0,0 +1,16 @@
+# https://leetcode.com/problems/evaluate-the-bracket-pairs-of-a-string/
+from typing import List
+
+
+# space complexity = O(n+m) n: len of s, m: len of knowledge (num of pairs)
+# m for creating dict + n for splitting input s
+# time complexity = O(n+m): for split list(n) find value in dict(m)
+def evaluate(self, s: str, knowledge: List[List[str]]) -> str:
+    str_dict = dict(knowledge)
+
+    split_list = s.split('(')
```

Comment:
I like the idea here, simple and nice implementation.
Although, this is `O(n)` for navigating through the `s` string, but needs to iterate the whole string twice.

Reply:
Could you please elaborate more on the "`O(n)` but iterating the whole string twice"?
Do you mean it with the implementation of `split`?

Reply:
Yes, even without knowing the actual implementation of split, we can imagine that split will iterate to the end of the string. But you're potentially calling split twice on the whole string, because you're doing split on the spitted string again one line 13.